database/buildSchema/43_views_functions_triggers.sql

```diff
@@ -700,6 +700,41 @@ CREATE TRIGGER review_assignment_validate_section_trigger
     WHEN (NEW.trigger IS NULL AND OLD.trigger IS NULL)
     EXECUTE FUNCTION public.enforce_asssigned_section_validity ();
 
+-- FUNCTION to set `available_sections` on given review_assignments based on other assignments
+CREATE OR REPLACE FUNCTION public.review_assignment_available_sections (assignment public.review_assignment)
+    RETURNS text[]
+    AS $$
+    SELECT
+        ARRAY ( WITH my_array AS (
+                SELECT DISTINCT
+                    (ts.code) available_sections
+                FROM
+                    template_section ts
+                    JOIN TEMPLATE t ON t.id = ts.template_id
+                    JOIN application a ON a.template_id = t.id
+                WHERE
+                    a.id = $1.application_id
+)
+                SELECT
+                    available_sections
+                FROM
+                    my_array
+                WHERE
+                    available_sections NOT IN (
+                        SELECT
+                            unnest(assigned_sections)
+                        FROM
+                            review_assignment
+                        WHERE
+                            status = 'ASSIGNED'
+                            AND stage_id = $1.stage_id
+                            AND level_number = $1.level_number
+                            AND application_id = $1.application_id
+                            AND id <> $1.id))
```

Comment:
I think you want to get rid of this last line -- otherwise it still says sections are available for the assigned reviewer, which is not correct.

Reply:
To also handle the allowed_sections, just add a little bit to the final WHERE clause:
```
 WHERE
                    available_sections NOT IN (
                        SELECT
                            unnest(assigned_sections)
                        FROM
                            review_assignment
                        WHERE
                            status = 'ASSIGNED'
                            AND stage_id = $1.stage_id
                            AND level_number = $1.level_number
                            AND application_id = $1.application_id
                          )
                          AND (available_sections = ANY($1.allowed_sections)
                          OR $1.allowed_sections IS NULL)
                          )
```

Reply:
Okay, have taken care of all this -- seems to working very nicely now ;)